who-metrics-ui/src/components/RepositoriesTable.tsx

```diff
@@ -224,6 +225,26 @@ const defaultFilters: Filter = {
   },
 };
 
+// Helper for generating the csv blob
+const generateCSV = (data: Repo[]): Blob => {
+  const output = json2csv(data);
+  return new Blob([output], { type: 'text/csv' });
+};
+
+// Helper for downloading the csv
+// There are warnings for Opera Mini incompatibility, but only 0.07% of users use that browser
+const saveFile = async (contents: Blob) => {
+  const a = document.createElement('a');
```

Comment:
I wonder if we should include the `a` in the DOM always and then use a `ref` to update the properties on it. Right now I think this will generate a new `a` each time the button is clicked, yes?

Reply:
This appears to be the case! Let me refactor